### ISSUE TITLE
Add Feature: Make possible to explicitly specify currency when localising numbers.

### ISSUE THREAD

**Issue by bponomarenko** [kazupon/vue-i18n#301]
Hi. Thanks for this plugin, it works really well for us. However there is one use case, which can't be implemented with `vue-i18n` at the moment.

### Use case
We need to display a list of formatted amounts in different currencies.
Data structure may look like this:
```javascript
const transactions = [
  { amount: 10123.45, currency: 'EUR' },
  { amount: 23.00, currency: 'GBP' },
  { amount: 45.40, currency: 'USD' },
];
```

### Feature request
Make it possible to explicitly specify currency when localising numbers.

Current set of arguments for `$n()` function looks like this:
* `{number} value: required`
* `{Path | Object} key: optional`
* `{Locale | Object} locale: optional`

This makes it possible to extend the second `key: Object` argument by adding optional `currency` option, which will override default value from the `numberFormats` configuration.

#### Example of possible API
Number formats:
```javascript
const numberFormats = {
  'en-US': {
    currency: {
      style: 'currency',
      currency: 'USD',
      currencyDisplay: 'code'
    }
  }
}
```
 Template:
```html
<ul v-for="transaction in transactions">
  <li>{{ $n(transaction.amount, { key: 'currency', currency: transaction.currency }) }}</p>
  <!-- Or use it even as a fallback options:
  <li>{{ $n(transaction.amount, { style: 'currency', 'currency': transaction.currency, currencyDisplay:
 'code' }) }}</li>
  -->
</ul>
```
Expected result:
```html
<ul>
  <li>EUR 10,123.45</li>
  <li>GBP 23.00</li>
  <li>USD 45.40</li>
</ul>
```

**Comment by bponomarenko** [comment]
@kazupon I can create PR if you ok with proposed API.

**Comment by kazupon** [comment]
@bponomarenko I look forward to your PR. :)

**Comment by kazupon** [comment]
closed (merge with #305)
